crypto/x509/v3_purp.c

```diff
@@ -698,26 +697,25 @@ int X509_check_ca(X509 *x)
     return check_ca(x);
 }
 
-/* Check SSL CA: common checks for SSL client and server */
+/* Check SSL CA: common checks for SSL client and server. */
 static int check_ssl_ca(const X509 *x)
 {
-    int ca_ret;
-    ca_ret = check_ca(x);
+    int ca_ret = check_ca(x);
+
     if (!ca_ret)
         return 0;
-    /* check nsCertType if present */
-    if (ca_ret != 5 || x->ex_nscert & NS_SSL_CA)
-        return ca_ret;
-    else
+    /* Check nsCertType if present */
+    if (ca_ret == 5 && (x->ex_nscert & NS_SSL_CA) != 0)
```

Comment:
Should the `!=` be `==`?

Reply:
Oops, I tried to invert the comparison but did so only partially.
I've fixed this, further simplifying the return value calculation
taking into account that at this point we know that `ca_ret != 0`.